instrumentation/resque/lib/opentelemetry/instrumentation/resque/instrumentation.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Resque
+      # The Instrumentation class contains logic to detect and install the Resque instrumentation
+      class Instrumentation < OpenTelemetry::Instrumentation::Base
+        install do |_config|
+          require_dependencies
+          patch
+        end
+
+        present do
+          defined?(::Resque)
+        end
+
+        option :job_class_span_names, default: false, validate: :boolean
```

Comment:
sidekiq has this as `enable_job_class_span_names` I think that is less rubyish.  I'll update sidekiq if I don't get any pushback on this 😄

Reply:
I picked `enable_job_class_span_names` to mirror sidekiq actually, but as with the span name option I'm really happy to harmonize on whatever. Hopefully its not something people have to actually interact with frequently 😆

Reply:
>I picked enable_job_class_span_names to mirror sidekiq actually, but as with the span name option I'm really happy to harmonize on whatever.

I really wanted to shorten the config option name 🙏 

>Hopefully its not something people have to actually interact with frequently

Definitely.

Reply:
I think `job_class_span_names` is certainly fine then, if one of the goals is shorter config option names!

Reply:
🚲 shed: you could make this an enum like `propagation_style` below. E.g.
```ruby
option :span_naming, default: :queue, validate: ->(opt) { %I[job_class queue].include?(opt) }
```